Packaging

rmnldwg opened the pull request:
Hi Yoel 👋

I forked and edited your package a bit so that it's installable smoothly. I must admit I am not too familiar with `setup.cfg` and `pyproject.toml` (I still use the older `setup.py`), so I am not sure everything holds up to the standards, but it's installable now (tried on Ubuntu and Windows). 

I also added a `.gitignore` file and remove the pycache and stuff.

So far, running the GUIs *almost* works. Here's what I did:

1. clone the repo using `git clone <insert-repo-url-here>`
2. go into it with `cd <insert-repo-name-here>`
3. create a virtual environment with `python -m venv .venv`
4. activate that environment, depending on the used OS (e.g. Ubuntu: `source .venv/bin/activate`). I tried it on Ubuntu and Windows so far
5. install the package using `pip install .`
6. run the 2D interface using `python -m adaptfx.Interface_2D` (that one means "run the following python module as main program".)

All those steps work fine and the GUI opens. However, when I the enter some values and hit "compute", it crashed with the following error message:

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Program Files\Python310\lib\tkinter\__init__.py", line 1921, in __call__
    return self.func(*args)
  File "C:\Users\roman\repos\Adaptive-fractionation\adaptfx\Interface_2D.py", line 241, in compute_plan
    lbl_output.pack()
  File "C:\Program Files\Python310\lib\tkinter\__init__.py", line 2425, in pack_configure
    self.tk.call(
_tkinter.TclError: bad window path name ".!frame6"
```

Which might have to do with my Tkinter version... or with the OS... or with anything else related to windows and GUIs, I don't know 😅

Best
Roman

YoelPH commented:
Hi Roman

Thanks for your valuable time!
The edits seem to be good :)
Furthermore, I think that I found the problem for the Error message. It _should_ work now

rmnldwg commented:
Hi Yoel,

Because you changed the code quite a bit since I originally created this pull request, I updated them. First, I pulled your changes, the I rebased my branch `packaging` where I had made the commits onto the changes you made and solved the merge conflicts.

Now, you should be able to merge the pull request.

Btw.: The interfaces are running now, but the 2D interface does not output any numbers. But I think it isn't the most up-to-date version anyways, right?

Best,
Roman
